inductor opinfo property tests (eager equiv, batch invariance, run to run)

v0i0 opened the pull request:
- **initial draft of opinfo property tests**
- **inductor opinfo-based property tests for eager equivalence, determinism and batch invariance**

Fixes #ISSUE_NUMBER


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

v0i0 committed:
initial draft of opinfo property tests

v0i0 committed:
inductor opinfo-based property tests for eager equivalence, determinism and batch invariance

v0i0 committed:
fix expected failures

v0i0 committed:
lint

v0i0 committed:
update for silu; improve xfail & run2run

v0i0 committed:
update docs

v0i0 committed:
lint

eellison reviewed (approved):
Wooooo

v0i0 committed:
make xfail more compact

v0i0 commented on test/inductor/test_torchinductor_opinfo_properties.py:
```diff
@@ -0,0 +1,1297 @@
+        "inductor_default",
+        "eager_equivalence",
+        None,
+    ): "div has numerical differences on CUDA",
```
done!

v0i0 commented:
@pytorchbot merge

v0i0 committed:
update expected

v0i0 committed:
update expected

v0i0 committed:
set determinism / split-k for determinism / batch invariance properties

v0i0 committed:
fix

v0i0 committed:
fix

v0i0 committed:
disable more rocm tests

v0i0 committed:
unlimited recompile

v0i0 committed:
fix

jansel reviewed (changes requested):
Failing tests/lints?

v0i0 committed:
fix lints/tests

v0i0 committed:
remove to fix; just want to merge

v0i0 committed:
take out div for now too

v0i0 committed:
disable on rocm

v0i0 committed:
more fixes

v0i0 committed:
remove matmul

v0i0 committed:
Remove reciprocal from inductor_numerics xfails

PR #174751 fixed reciprocal to use float32 for division_rounding,
which makes these xfails stale under the inductor_numerics backend.
